config/routes.rb

```diff
@@ -1,6 +1,9 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  # root path is '/' or '/users'
+  root 'users#index'
 
-  # Defines the root path route ("/")
-  # root "articles#index"
+  get '/users', to: 'users#index'
+  get '/users/:id', to: 'users#show'
+  get '/users/:id/posts', to: 'posts#index'
+  get '/users/:id/posts/:id', to: 'posts#show'
```

Comment:
## Optional
- [ ] I suggest that you use `Resource routing` because it allows you to quickly declare all of the common routes for a given resourceful controller
```
resources:users, only: [: index, : show] do
    resources:posts, only: [: index, : show]
  end
```

Reply:
thank you so much i will do it in next